app/handlers/tests/test_handlers_connector.py

```diff
@@ -395,3 +401,46 @@ def test_not_past_due_annotation(
         assert run.patch_mock.meta == {}
         run.k8s_client_mock.patch_namespaced_pod.assert_not_called()
         mock_get_image.assert_not_called()
+
+
+class TestGetConnectorPod:
```

Comment:
It'd be nice to cover other spec fields logic as well for completeness. I can do it in a separate PR.